units/spawnhost_termination_alert.go

```diff
@@ -0,0 +1,102 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/user"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+)
+
+const (
+	spawnhostTerminationAlertName = "spawnhost-termination-alert"
+
+	emailSubject = "Spawn Host Terminated"
+	emailBody    = "Your spawn host '%s' has been terminated by Evergreen because it failed to provision. " +
+		"Feel free to spawn another."
+)
+
+func init() {
+	registry.AddJobType(spawnhostTerminationAlertName, func() amboy.Job {
+		return makeSpawnhostTerminationAlertJob()
+	})
+}
+
+type spawnhostTerminationAlertJob struct {
+	job.Base `bson:"base" json:"base" yaml:"base"`
+
+	User string `bson:"user"`
+	Host string `bson:"host"`
+
+	sender send.Sender
```

Comment:
It might be better to put an evergreen.Environment here, and initialise it with mock.Environment.

The sender provided by the mock environment is an internal sender, that is public so you can use it for intropsection